### PR TITLE
Resetting of high scores, scheduler and optimizer for fine-tuning/domain adaptation

### PR DESCRIPTION
**Pull request by juliakreutzer**
In the case of fine-tuning or domain adaptation, one might want to overwrite the existing scheduler or optimizer or the previously tracked high score. 
This is specifically relevant when fine-tuning with a different metric or a data where the validation score is in a different range than the loaded model (e.g. pre-training on BLEU and fine-tuning on PPL would result in never storing a new checkpoint). When the data is much smaller than the previous training data, one might want to reduce the patience.

We can also think about modeling this as "load_x_from_ckpt" rather than "reset_x", but for now, it seems that in the default case we want to load everything.

### CODE REVIEW

**Comment by juliakreutzer** [comment]
Awesome, thanks for confirming this so quickly @dwhitena! And an official welcome as collaborator! :tada:
